qemu/tests/migration_with_boot_pattern.py

```diff
@@ -0,0 +1,84 @@
+import aexpect
+import logging
+import time
+
+from autotest.client.shared import utils
+from autotest.client.shared import error
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    KVM migration test:
+    1) Start VM
+    2) Wait in background for the occurrence of the boot_pattern while
+       migrating the VM in a loop
+    3) Issue one more migration after the pattern is reached
+    4) Optionally send a line to serial port (to proceed booting) followed
+       by yet another migration
+    5) Login to the machine
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    def wait_for_boot_pattern(boot_pattern, vm):
+        """
+        Waits till last_line_matches and survives vm-migration
+        """
+        patterns = [boot_pattern]
+        while True:
+            try:
+                if vm.serial_console.read_until_last_line_matches(patterns,
+                                                                  timeout=1):
+                    break
+            except aexpect.ExpectError:
+                pass
+
+    vm = env.get_vm(params["main_vm"])
+    mig_timeout = float(params.get("mig_timeout", "3600"))
+    mig_protocol = params.get("migration_protocol", "tcp")
+    mig_cancel_delay = int(params.get("mig_cancel") == "yes") * 2
+    boot_pattern_timeout = float(params.get("boot_pattern_timeout", 3600))
+    boot_pattern = params["boot_pattern"]
+    post_boot_pattern_cmd = (params.get("post_boot_pattern_cmd")
+                             .decode("unicode_escape"))
+    login_timeout = int(params.get("login_timeout", 360))
```

Comment:
Not suggest to customize `login_timeout` if there is no special needs.

Reply:
Well this test is supposed to be used for boot-pattern checks. It might be useful to change the boot timeout __after__ the boot pattern is hit (it could be later in the boot process and we might want shorter timeout, or even longer one than default).